pkg/util/containers/filter.go

```diff
@@ -72,37 +79,62 @@ type Filter struct {
 	NamespaceExcludeList []*regexp.Regexp
 }
 
+// filterErrors holds a list of regex errors and warnings for invalid filters
+type filterErrors struct {
+	errors map[string]struct{}
+	sync.Mutex
+}
+
 var sharedFilter *Filter
+var filterErrs *filterErrors
 
-func parseFilters(filters []string) (imageFilters, nameFilters, namespaceFilters []*regexp.Regexp, err error) {
+func parseFilters(filters []string) (imageFilters, nameFilters, namespaceFilters []*regexp.Regexp, filterErrs []string, err error) {
+	var filterWarnings []string
 	for _, filter := range filters {
 		switch {
-		case strings.HasPrefix(filter, "image:"):
-			pat := strings.TrimPrefix(filter, "image:")
-			r, err := regexp.Compile(strings.TrimPrefix(pat, "image:"))
+		case strings.HasPrefix(filter, imageFilterPrefix):
+			r, err := validateFilter(filter, imageFilterPrefix)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("invalid regex '%s': %s", pat, err)
+				filterErrs = append(filterErrs, err.Error())
+				continue
 			}
 			imageFilters = append(imageFilters, r)
-		case strings.HasPrefix(filter, "name:"):
-			pat := strings.TrimPrefix(filter, "name:")
-			r, err := regexp.Compile(pat)
+		case strings.HasPrefix(filter, nameFilterPrefix):
+			r, err := validateFilter(filter, nameFilterPrefix)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("invalid regex '%s': %s", pat, err)
+				filterErrs = append(filterErrs, err.Error())
+				continue
 			}
 			nameFilters = append(nameFilters, r)
-		case strings.HasPrefix(filter, "kube_namespace:"):
-			pat := strings.TrimPrefix(filter, "kube_namespace:")
-			r, err := regexp.Compile(pat)
+		case strings.HasPrefix(filter, kubeNamespaceFilterPrefix):
+			r, err := validateFilter(filter, kubeNamespaceFilterPrefix)
 			if err != nil {
-				return nil, nil, nil, fmt.Errorf("invalid regex '%s': %s", pat, err)
+				filterErrs = append(filterErrs, err.Error())
+				continue
 			}
 			namespaceFilters = append(namespaceFilters, r)
 		default:
-			log.Warnf("Container filter %q is unknown, Ignoring it. The supported filters are 'image', 'name' and 'kube_namespace'", filter)
+			warnmsg := fmt.Sprintf("Container filter %q is unknown, ignoring it. The supported filters are 'image', 'name' and 'kube_namespace'", filter)
+			log.Warnf(warnmsg)
+			filterWarnings = append(filterWarnings, warnmsg)
+
 		}
 	}
-	return imageFilters, nameFilters, namespaceFilters, nil
+	if len(filterErrs) > 0 {
+		return nil, nil, nil, append(filterErrs, filterWarnings...), errors.New(filterErrs[0])
+	}
+	return imageFilters, nameFilters, namespaceFilters, filterWarnings, nil
+}
+
+// validateFilter checks a filter's regex
+func validateFilter(filter string, filterPrefix string) (*regexp.Regexp, error) {
```

Comment:
I liked the idea of having this block in a separate function instead of repeating it multiple times in `parseFilters`, thank you!

Nit about the function name though: The block doesn't really validate the filter, it just tries to extract a `*regexp.Regexp`. I think we can rename the function `filterToRegex` or something like that.